includes/woocommerce-memberships/class-limit-purchase.php

```diff
@@ -105,4 +115,48 @@ public static function woocommerce_cart_product_cannot_be_purchased_message( $me
 		}
 		return $message;
 	}
+
+	/**
+	 * Get user from email.
+	 *
+	 * @return false|int User ID if found by email address, false otherwise.
+	 */
+	private static function get_user_id_from_email() {
+		$billing_email = filter_input( INPUT_POST, 'billing_email', FILTER_SANITIZE_EMAIL );
+		if ( $billing_email ) {
+			$customer = \get_user_by( 'email', $billing_email );
+			if ( $customer ) {
+				return $customer->ID;
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Validate network subscription for logged out readers.
+	 *
+	 * @param array     $data   Checkout data.
+	 * @param WC_Errors $errors Checkout errors.
+	 */
+	public static function validate_network_subscription( $data, $errors ) {
+		if ( is_user_logged_in() || ! function_exists( 'WC' ) ) {
+			return;
+		}
+		$id_from_email = self::get_user_id_from_email();
+		if ( $id_from_email ) {
+			$cart_items = WC()->cart->get_cart();
+			foreach ( $cart_items as $cart_item ) {
+				$product = $cart_item['data'];
+				$network_active_subscription = self::get_network_equivalent_subscription_for_current_user( $product, $id_from_email );
+				if ( $network_active_subscription ) {
+					$error_message = sprintf(
+						/* translators: %s: Site URL */
+						__( "You can't buy this subscription because you already have it active on %s", 'newspack-network' ),
```

Comment:
I'm fine with this updated error message, but just wanted to share why I removed the site name initially. Since this is for logged out readers, its possible for someone to input another persons email address here, in which case they'd learn about where another person's subscription exists.

This is probably not a big deal, but wanted to share just in case.